### PR TITLE
WT-5775 Fix a leak of update structures in the history store.

### PR DESCRIPTION
**Pull request by michaelcahill**

### CODE REVIEW

**Comment by tetsuo-cpp** [comment]
@michaelcahill Had a chat with @kommiharibabu and we think this change needs to go in rollback to stable too.

File: src/history/hs.c [hunk 1]
```diff
@@ -359,6 +360,15 @@ __hs_insert_record_with_btree_int(WT_SESSION_IMPL *session, WT_CURSOR *cursor, W
     /* Search the page and insert the mod list. */
     WT_WITH_PAGE_INDEX(session, ret = __wt_row_search(cbt, &cursor->key, true, NULL, false, NULL));
     WT_ERR(ret);
+    /* We expect this to be a new insert into the history store, but cope if we find existing
+     * updates. */
```
Comment (agorrod) [inline src/history/hs.c]: We still do multi-line comments as a block comment please.

**Comment by agorrod** [comment]
I'm happy for this to be merged in if testing passes.

**Comment by michaelcahill** [comment]
@tetsuo-cpp, for the fix to rollback_to_stable, do you and @kommiharibabu want me to fix that in this ticket or open a separate ticket for that?

**Comment by kommiharibabu** [comment]
@michaelcahill If this pull request is not in urgent to merge and the fix is small, I feel it can be done as part of this pull request, otherwise a new ticket is better.

**Comment by michaelcahill** [comment]
@agorrod, I've refactored and updated the change to cover rollback_to_stable as well.  Can you please do a (hopefully final) review?

File: src/history/hs.c [hunk 2]
```diff
@@ -261,6 +261,39 @@ __wt_hs_cursor_close(WT_SESSION_IMPL *session, uint32_t session_flags, bool is_o
...
+__wt_hs_modify(WT_CURSOR_BTREE *hs_cbt, WT_UPDATE *hs_upd)
+{
+    WT_DECL_RET;
+    WT_PAGE_MODIFY *mod;
+    WT_SESSION_IMPL *session;
+    WT_UPDATE *last_upd;
+
+    session = (WT_SESSION_IMPL *)hs_cbt->iface.session;
+
+    /* Cope if we find existing updates. */
```
Comment (agorrod) [inline src/history/hs.c]: Would mind asserting somewhere that there is no existing entry in the insert case?

Comment (michaelcahill) [reply src/history/hs.c]: Sure, done in ef8bf5dab.

**Pull request merged**